airflow/models/taskinstance.py

```diff
@@ -299,11 +299,14 @@ def clear_task_instances(
         )
         dag_run_state = DagRunState(dag_run_state)  # Validate the state value.
         for dr in drs:
-            dr.state = dag_run_state
-            dr.start_date = timezone.utcnow()
-            if dag_run_state == DagRunState.QUEUED:
-                dr.last_scheduling_decision = None
-                dr.start_date = None
+            # Otherwise task(s) cleared in the middle of the running DagRun.
+            # There is no need to reset anything
+            if dr.state in [State.SUCCESS, State.FAILED]:
```

Comment:
nit, better use a tuple or set here instead of a list :) ideally we also create the set outside of the loop only once.

Reply:
moved to `State` so everybody can reuse it now :)